tests/test_behavior/test_video_interface.py

```diff
@@ -1,290 +1,314 @@
-        self.metadata = self.nwb_converter.get_metadata()
-        self.metadata["NWBFile"].update(session_start_time=datetime.now(tz=gettz(name="US/Pacific")))
-        self.nwbfile_path = self.test_dir / "video_test.nwb"
-        self.aligned_segment_starting_times = [0.0, 50.0]
-
-    def tearDown(self) -> None:
-        shutil.rmtree(self.test_dir)
-        del self.nwb_converter
-
-    def create_videos(self):
-        video_file1 = str(self.test_dir / "test1.avi")
-        video_file2 = str(self.test_dir / "test2.avi")
-        video_file3 = str(self.test_dir / "test3.avi")
-        number_of_frames = 30
-        number_of_rows = 640
-        number_of_columns = 480
-        frameSize = (number_of_columns, number_of_rows)  # This is give in x,y images coordinates (x is columns)
-        fps = 25
-        # Standard code for specifying image formats
-        fourcc_specification = ("M", "J", "P", "G")
-        # Utility to transform the four code specification to OpenCV specification
-        fourcc = cv2.VideoWriter_fourcc(*fourcc_specification)
-
-        writer1 = cv2.VideoWriter(
-            filename=video_file1,
-            fourcc=fourcc,
-            fps=fps,
-            frameSize=frameSize,
-        )
-        writer2 = cv2.VideoWriter(
-            filename=video_file2,
-            fourcc=fourcc,
-            fps=fps,
-            frameSize=frameSize,
-        )
-        writer3 = cv2.VideoWriter(
-            filename=video_file3,
-            fourcc=fourcc,
-            fps=fps,
-            frameSize=frameSize,
-        )
 
-        for frame in range(number_of_frames):
-            writer1.write(np.random.randint(0, 255, (number_of_rows, number_of_columns, 3)).astype("uint8"))
-            writer2.write(np.random.randint(0, 255, (number_of_rows, number_of_columns, 3)).astype("uint8"))
-            writer3.write(np.random.randint(0, 255, (number_of_rows, number_of_columns, 3)).astype("uint8"))
-
-        writer1.release()
-        writer2.release()
-        writer3.release()
-
-        return [video_file1, video_file2, video_file3]
-
-    def create_video_converter(self):
-        class VideoTestNWBConverter(NWBConverter):
-            data_interface_classes = dict(
-                Video1=VideoInterface,
-                Video2=VideoInterface,
-            )
-
-        source_data = dict(
-            Video1=dict(
-                file_paths=self.video_files[0:2],
-                metadata_key_name="Video1",
-            ),
-            Video2=dict(
-                file_paths=[self.video_files[2]],
-                metadata_key_name="Video2",
-            ),
+# Common fixtures for all tests
+@pytest.fixture
+def cv2():
+    """Import and return cv2 or skip the test if not installed."""
+    return pytest.importorskip("cv2")
+
+
+@pytest.fixture
+def test_dir(tmp_path):
+    """Create and return a test directory using pytest's tmp_path fixture."""
+    return tmp_path
+
+
+@pytest.fixture
```

Comment:
What matter is that this is well scoped so the computation is not repeated:

https://docs.pytest.org/en/6.2.x/fixture.html#scope-sharing-fixtures-across-classes-modules-packages-or-session